### PR TITLE
Add option for adding arbitrary headers to HTTP operators

### PR DESCRIPTION
**Pull request by alexpogue**
Add "extraHeaders" parameter which takes a list of Strings in a multivalued format as follows
```
extraHeaders: "X-Header1: value1", "X-Header2: value2"
                         ^                    ^
zero or more spaces    here                 and here
```

### CODE REVIEW

File: com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java [hunk 1]
```diff
@@ -175,6 +181,14 @@ else if(headerContentType.equals(MIME_JSON)){
 			req.setParams(tuple.getObject(schema.getAttribute(0).getName()).toString());
 		}
 
+		for(String header : extraHeaders) {
+			String[] headerParts = header.split(":\\s*", 2);
```
Comment (alexpogue) [inline com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java]: `"\\s*"` is regex that matches zero or more spaces.

File: com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java [hunk 2]
```diff
@@ -175,6 +181,14 @@ else if(headerContentType.equals(MIME_JSON)){
...
+			String headerName = headerParts[0];
+			String headerValue = headerParts[1];
+			System.out.println("Seeing headerName=" + headerName + ", headerValue=" + headerValue);
```
Comment (ddebrunner) [inline com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java]: Need to remove this System.out message, or make it a trace

File: com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPStreamReader.java [hunk 3]
```diff
@@ -180,10 +185,19 @@ public void initialize(OperatorContext op) throws Exception {
 		if(authenticationFile != null) {
             authenticationFile = authenticationFile.trim();
         }
+		Map<String, String> extraHeaderMap = new HashMap<String, String>();
+		for(String header : extraHeaders) {
+			String[] headerParts = header.split(":\\s*", 2);
+			String headerName = headerParts[0];
+			String headerValue = headerParts[1];
+			//trace.log(TraceLevel.ERROR, "headerName = " + headerName + ", headerValue = " + headerValue);
```
Comment (ddebrunner) [inline com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPStreamReader.java]: Remove this line or make it a real trace.

Comment (alexpogue) [inline com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPStreamReader.java]: Fixed

File: com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java (hunk 2 above)
Comment (alexpogue) [inline com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java]: Good catch, fixed

**Pull request merged**